assets/app/scripts/services/applicationGenerator.js

```diff
@@ -88,7 +88,15 @@ angular.module("openshiftConsole")
       var service = scope._generateService(input, input.name, ports);
       if (service) {
         resources.service = service;
+
         // Only attempt to generate a route if there is a service.
+        // Use the port name from the service when generating the route.
+        if (input.routing.targetPort) {
+          var servicePort = _.find(service.spec.ports, function(servicePort) {
+            return servicePort.targetPort === input.routing.targetPort.port;
+          });
+          input.routing.targetPort.port = servicePort.name;
```

Comment:
I don't really like targetPort.port starting out as a number and ending up as a name... any way to make that clearer in the code?